### PR TITLE
CMSSW Config Tracing Tool

### PR DESCRIPTION
**Pull request by schneiml**
This PR adds two tools that help with understanding CMSSW configurations.

- `cmsswFiletrace.py` uses the Python `settrace` feature to track call graphs and therefore imports etc.
- `cmsswConfigtrace.py` uses hooks to attach additional information to the config objects, like the `EnablePSetHistory` thing used to do.

`cmsswConfigtrace` is more advanced and comes with a small web frontend to read the saved information.

I put the tools into `DQMOffline/Configuration` since I intend to use them for DQM offline configuration, though they are rather generic.

### CODE REVIEW

**Comment by andrius-k** [comment]
please test

**Comment by andrius-k** [comment]
+1

File: DQMOffline/Configuration/scripts/cmsswConfigtrace.py [hunk 1]
```diff
@@ -0,0 +1,604 @@
+#!/usr/bin/env python2
```
Comment (fabiocos) [inline DQMOffline/Configuration/scripts/cmsswConfigtrace.py]: @schneiml why python2 and not just python?

File: DQMOffline/Configuration/scripts/cmsswFiletrace.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python2
```
Comment (fabiocos) [inline DQMOffline/Configuration/scripts/cmsswFiletrace.py]: @schneiml same comment here

**Comment by fabiocos** [comment]
@schneiml could you please clarify? I would like to move forward with this PR, it is just a minor technical question

**Comment by schneiml** [comment]
@fabiocos I changed the python lines.
BTW, a experimental setup of the UI can be seen here: http://cmsswconfigexplore.cern.ch:1234/ (only inside CERN)

**Comment by andrius-k** [comment]
please test

**Comment by andrius-k** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Pull request merged**